R/infrastructure.R

```diff
@@ -33,6 +39,12 @@ use_testthat <- function(pkg = ".") {
     pkg = pkg
   )
 
+  if (use_catch) {
+    message("Adding framework for Catch unit tests")
+    add_desc_package(pkg, "LinkingTo", "testthat")
+    use_template("test-example.cpp", "src/test-example.cpp")
+    use_template("test-runner", "src/test-runner.cpp")
+  }
```

Comment:
I think you also need the `R` template file: https://github.com/hadley/testthat/blob/master/R/test-compiled-code.R#L183-L187